ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -310,16 +310,19 @@ class ClientQueryViewController: UITableViewController, Themeable {
 							lastTappedItemLocalID = rowItem.localID
 
 							core.downloadItem(rowItem, options: [ .returnImmediatelyIfOfflineOrUnavailable : true ]) { [weak self, query] (error, core, item, _) in
+
+								guard let `self` = self else { return }
 								OnMainThread {
 									if (error == nil) || (error as NSError?)?.isOCError(withCode: .itemNotAvailableOffline) == true {
-										if let item = item, item.localID == self?.lastTappedItemLocalID, let core = core {
-											let itemViewController = DisplayHostViewController(for: item, with: core, root: query.rootItem!)
-											self?.navigationController?.pushViewController(itemViewController, animated: true)
+										if let item = item, item.localID == self.lastTappedItemLocalID, let core = core {
```

Comment:
@pablocarmu I would probably not mix` if let` constructs and regular `if` conditions

Reply:
I'll split them into 2 if-else